src/csharp/Grpc.Core/Internal/GrpcThreadPool.cs

```diff
@@ -82,7 +85,8 @@ public void Start()
 
                 for (int i = 0; i < poolSize; i++)
                 {
-                    threads.Add(CreateAndStartThread(i));
+                    var optionalProfiler = i < threadProfilers.Count ? threadProfilers[0] : null;
```

Comment:
not sure exactly where `threadProfilers` are set, is looks like the list of `threadProfilers` should that be empty at this point? And should each thread be getting `threadProfilers[i]`?